contracts/.solhint.json

```diff
@@ -3,11 +3,10 @@
   "plugins": ["prettier"],
   "rules": {
     "code-complexity": ["error", 7],
-    "compiler-version": ["error", "^0.6.12"],
+    "compiler-version": ["warn", "^0.6.12"],
```

Comment:
_(arbitrary comment location to start a thread)_

For Arbitrum and Optimism, let's make sure to test the advanced mode "send to standard pubkey" functionality using both EOAs that have sent transactions in the past and new EOAs

Reply:
> functionality using both EOAs that have sent transactions in the past and new EOAs

When you say "new EOAs" here do you mean making sure that EOAs that have never sent a tx cannot receive an advanced mode tx? If so, I confirmed this is the case.

I also tested Advanced Mode on both networks with an account that sent its first tx on both networks today. Both worked as expected without issue!

Reply:
> I also tested Advanced Mode on both networks with an account that sent its first tx on both networks today. Both worked as expected without issue!

Ok great! We should also test advanced mode with an EOA that has sent txs as long ago as possible, without recent ones, to test the opposite case (in case there's new/old tx types on either network)—and then this would cover the "EOA tests" my unclear comment referred to. This may not be too practical depending on our past usages

Reply:
I tested out what happens if you try to send to an account that has never sent a transaction, except for receiving funds over the bridge. (0x22222B7C879e6c09690d1E5Eb10437E68987882c)

On Optimism, it simply says this account has never done a txn, which is good.

On Arbitrum, you get this validation error:

<img width="535" alt="Screen Shot 2022-04-01 at 5 04 16 PM" src="https://user-images.githubusercontent.com/1479802/161341314-d2c3db32-500c-4829-9f0b-4feb96f05ab2.png">

As long as we feel confident this will always be what happens for such addresses, I think it's fine. It's an edge case, and users are not able to send funds so there's no harm. What do you think @mds1?

Reply:
Just pushed ad9f879, so pub key recovery will now work on Arbitrum. Details on the limitations and future improvements to remove those limitations is being tracked in https://github.com/ScopeLift/umbra-protocol/issues/320

Reply:
Should be fixed with 1df8b62